dds/DCPS/DiscoveryBase.h

```diff
@@ -957,6 +997,164 @@ namespace OpenDDS {
+
+          Consequently, all of the code up to the call to match_continue should be moved to the first part of match_continue.
+          The first part of match continue should be the consistency check that was being done in TopicDetails for non-xtypes or the new consistency check for extypes.
+          You will probably need to resurrect the counter that was in TopicDetails.
+
+          If all of the data used for computing consistency is immutable, i.e.,
+            - the topic name
+            - the type name
+            - the type identifiers
+            - etc.
+          Then you are done.
+          Otherwise, we have to think about cases where a local/discovered changes in a way that alters consistency.
+          If we do have to support this, then it will have to be in future work.
+         */
+
+        // 1. collect type info about the writer, which may be local or discovered
+        XTypes::TypeInformation* writer_type_info = 0;
+
+        const LocalPublicationIter lpi = local_publications_.find(writer);
+        DiscoveredPublicationIter dpi;
+        bool writer_local = false, already_matched = false;
+        if (lpi != local_publications_.end()) {
+          writer_local = true;
+          writer_type_info = &lpi->second.type_info_;
+        } else if ((dpi = discovered_publications_.find(writer))
+                   != discovered_publications_.end()) {
+          writer_type_info = &dpi->second.type_info_;
+        } else {
+          return; // Possible and ok, since lock is released
+        }
+
+        // 2. collect type info about the reader, which may be local or discovered
+        XTypes::TypeInformation* reader_type_info = 0;
+
+        const LocalSubscriptionIter lsi = local_subscriptions_.find(reader);
+        DiscoveredSubscriptionIter dsi;
+        bool reader_local = false;
+        if (lsi != local_subscriptions_.end()) {
+          reader_local = true;
+          reader_type_info = &lsi->second.type_info_;
+        } else if ((dsi = discovered_subscriptions_.find(reader))
+                   != discovered_subscriptions_.end()) {
+          reader_type_info = &dsi->second.type_info_;
+        } else {
+          return; // Possible and ok, since lock is released
+        }
+
+        MatchingData md;
+
+        // if the type object is not in cache, send RPC request
+        md.writer = writer;
+        md.reader = reader;
+        md.time_added_to_map = MonotonicTimePoint::now();
+
+        if ((writer_type_info->minimal.typeid_with_size.type_id.kind() != XTypes::TK_NONE) &&
+            (reader_type_info->minimal.typeid_with_size.type_id.kind() != XTypes::TK_NONE)) {
+          if (!writer_local && reader_local) {
+            if (type_lookup_service_ && !type_lookup_service_->type_object_in_cache(writer_type_info->minimal.typeid_with_size.type_id)) {
+              XTypes::TypeIdentifierSeq type_ids;
+              type_ids.append(writer_type_info->minimal.typeid_with_size.type_id);
```

Comment:
`type_ids` isn't used and should be removed.

Reply:
@kuznetsovmoci Assuming you agree. I'll remove.